### ISSUE TITLE
Add import qualifications doesn't work.

### ISSUE THREAD

**Issue by kofno** [FrigoEU/psc-ide-vim#3]
I can remove import qualifications. And adding the type signature works. And the syntastic checking works. But, for some reason, I can't get import qualifications to work.

**Comment by FrigoEU** [comment]
Hey, psc-ide-vim does quite some string parsing etc, so could you give me the exact line/file that you're working with, and which command isn't working?

**Comment by kofno** [comment]
```
module Main where

import Prelude
import Control.Monad.Eff
import Control.Monad.Eff.Console

main :: forall e. Eff (console :: CONSOLE | e) Unit
main = do
  log "Hello sailor!"
```

PSCIDEaddImportQualifications doesn't work, just silently does nothing.
PSCIDEapplySuggestion doesn't work. 'No suggestion found'

You can see in this screen shot that suggestions were detected:

<img width="579" alt="screenie" src="https://cloud.githubusercontent.com/assets/19907/14112950/da6f7df8-f59e-11e5-9ad1-d3d5d647e48e.png">

**Comment by FrigoEU** [comment]
Hmm, very strange, could you tell me what version of psc-ide-vim, psc/psc-ide and pulp you're using?

**Comment by kofno** [comment]
pulp version 8.1
psc-ide-vim is at ff6d0127517ad16012b 
psc is 0.8.3.0
psc-ide-client is 0.8.3.0

It looks like I have an older psc-ide in the path somewhere too, which is 0.6.1

EDIT:
removing 0.6.1 psc-ide didn't make a difference.

**Comment by FrigoEU** [comment]
Can't seem to be able to reproduce this so far. Can you do this:

  let g:psc_ide_log_level = 3

and try it again? You should see some more logging of what psc-ide-vim is trying to do, maybe that'll help us pinpoint the problem.

**Comment by kofno** [comment]
Sorry for taking so long to get back to this.

With the logging turned up, I see this. Then, when I hit ENTER, it goes away, but the buffer is never updated.

```
Adding import qualifications to : [19, 18, 17, 16, 14, 13, 11, 10, 9, 7]
PSCIDEapplySuggestion: lineNr: 19
PSCIDEapplySuggestion: BufferNr: 1
PSCIDEapplySuggestion: lineNr: 18
PSCIDEapplySuggestion: BufferNr: 1
PSCIDEapplySuggestion: lineNr: 17
PSCIDEapplySuggestion: BufferNr: 1
PSCIDEapplySuggestion: lineNr: 16
PSCIDEapplySuggestion: BufferNr: 1
PSCIDEapplySuggestion: lineNr: 14
PSCIDEapplySuggestion: BufferNr: 1
PSCIDEapplySuggestion: lineNr: 13
PSCIDEapplySuggestion: BufferNr: 1
PSCIDEapplySuggestion: lineNr: 11
PSCIDEapplySuggestion: BufferNr: 1
PSCIDEapplySuggestion: lineNr: 10
PSCIDEapplySuggestion: BufferNr: 1
PSCIDEapplySuggestion: lineNr: 9
PSCIDEapplySuggestion: BufferNr: 1
PSCIDEapplySuggestion: lineNr: 7
PSCIDEapplySuggestion: BufferNr: 1
```

When I try PSCIDEapplySuggestion, I see this:

```
PSCIDEapplySuggestion: lineNr: 19
PSCIDEapplySuggestion: BufferNr: 1
PSCIDEapplySuggestion: No suggestion found
```

**Comment by jazmit** [comment]
I'm seeing the exact same issue, using neovim

```
$ nvim --version
NVIM 0.1.3
$ pulp --version
Pulp version 8.1.1
psc version 0.8.4.0 using /usr/bin/psc
$ psc-ide-client --version
0.8.4.0
psc-ide-vim: ff6d0127517ad16012b27787fb1027f8ae8e99f0
```

**Comment by jazmit** [comment]
OK, I did a little more digging and found a work-around.
If you have Syntastic populate and open the location list with `:Errors`, everything then works as designed.  I'm no vimscript expert, but could this be a neovim issue where you are retrieving the  location list before it is populated in https://github.com/FrigoEU/psc-ide-vim/blob/master/ftplugin/purescript_pscide.vim line 247?

**Comment by jazmit** [comment]
More digging, and I found this in the syntastic help

```
By default syntastic doesn't fill the |location-list| with the errors found by
the checkers, in order to reduce clashes with other plugins. Consequently, if
you run `:lopen` or `:lwindow` rather than `:Errors` to open the error window
you wouldn't see syntastic's list of errors. If you insist on using `:lopen`
or `:lwindow` you should either run `:SyntasticSetLoclist` after running the
checks, or set |'syntastic_always_populate_loc_list'| which tells syntastic to
update the |location-list| automatically.
```

This means adding the following to vimrc fixes the problem:
`au FileType purescript let syntastic_always_populate_loc_list=1`

**Comment by FrigoEU** [comment]
Thanks a lot for the research! `au FileType purescript let syntastic_always_populate_loc_list=1` is definitely the problem. I have this enabled by default so I never ran into this issue. I'll do some testing to see if I can do the suggestions some different way, otherwise I'll add this to the readme. Thanks again!

**Comment by kritzcreek** [comment]
that's some detective conan level digging!

**Comment by FrigoEU** [comment]
Got it. I'll release this soon, together with the automatic imports feature from purescript 8.4.

**Comment by kofno** [comment]
\o/

Thanks everyone! You are awesome!

**Comment by FrigoEU** [comment]
Should be fixed in 0.3.0 (just uploaded now). Also added :PSCIDEimportIdentifier. Let me know if it's not fixed for you yet, otherwise I'll close this soon.

**Comment by jazmit** [comment]
Works like a dream!  Tried out `:PSCIDEImportIndentifier`, and it is a life saver.  I just started using Purescript a couple of days ago and I was wondering how on earth it's possible to get anything done if you have to do all those imports manually...

**Comment by FrigoEU** [comment]
Excellent. After using importIdentifier and applySuggestion for a while, I find I'm rarely editing my import list manually. Most of the time I have it folded automatically with a fold expression. Only when I need to resolve some conflict or add an operator (still need to fix importIdentifier so it can also handle operators) do I switch to manual folding and edit by hand, but it happens rarely.

**Comment by FrigoEU** [comment]
Fixed by 792c56a07d89f3c2e0a096b062d80a1ccfa07641.
